apps/ex_nvr/lib/ex_nvr/events.ex

```diff
@@ -0,0 +1,71 @@
+defmodule ExNVR.Events do
+  alias Membrane.Event
+  alias ExNVR.Model.{Device, Event}
+  alias Ecto.Multi
+  alias ExNVR.Repo
+
+  def create(params, device, "lpr") do
+    formated_event =
+      params
+      |> rename_milesight_keys(device)
+      |> Map.put(:type, "lpr")
+      |> Map.put(:device_id, device.id)
+
+    Multi.new()
+    |> Multi.insert(:event, Event.changeset(formated_event))
+    |> Multi.run(:create_thumbnail, fn _repo, %{event: event} ->
```

Comment:
A better approach to save the thumbnails is to put them on the argument of the function
```elixir
def create_lpr_event(device, params, plate_picture \\ nil, full_picture \\ nil) do
end
```
No need to use transaction here, even if the thumbnails are not saved it's not a problem, we only need the event.